### PR TITLE
[dag] chain health backoff backpressure

### PR DESCRIPTION
**Pull request by ibalajiarun**
### Description

This PR implements the chain health based backpressure for the DAG. It uses the existing ChainHealthBackoff configs and implementations and wires it into the DAG Driver and Round State.

### CODE REVIEW

**Comment by ibalajiarun** [comment]
Current dependencies on/for this PR:
* main
  * **PR #10689** <a href="https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10689" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/aptos-labs/aptos-core/10689?utm_source=stack-comment).

File: consensus/src/dag/dag_driver.rs [hunk 1]
```diff
@@ -290,6 +286,36 @@ impl DagDriver {
...
+            } else {
+                (
+                    self.payload_config.max_sending_txns_per_round,
+                    self.payload_config.max_sending_size_per_round_bytes,
+                )
+            };
+        // TODO: warn/panic if division yields 0 txns
+        let max_txns = max_txns_per_round
+            .saturating_div(self.epoch_state.verifier.len() as u64)
+            .max(1);
```
Comment (sasha8) [inline consensus/src/dag/dag_driver.rs:312]: Why max 1?

File: consensus/src/dag/dag_driver.rs [hunk 2]
```diff
@@ -290,6 +286,36 @@ impl DagDriver {
...
+        let max_txn_size_bytes = max_size_per_round_bytes
+            .saturating_div(self.epoch_state.verifier.len() as u64)
+            .max(1024);
```
Comment (sasha8) [inline consensus/src/dag/dag_driver.rs:315]: why max 1024?

File: consensus/src/dag/dag_driver.rs [hunk 3]
```diff
@@ -290,6 +286,36 @@ impl DagDriver {
...
+                )
+            } else {
+                (
+                    self.payload_config.max_sending_txns_per_round,
+                    self.payload_config.max_sending_size_per_round_bytes,
+                )
+            };
+        // TODO: warn/panic if division yields 0 txns
+        let max_txns = max_txns_per_round
+            .saturating_div(self.epoch_state.verifier.len() as u64)
```
Comment (sasha8) [inline consensus/src/dag/dag_driver.rs:311]: This is not accurate. If we are here then we have validators behind. So to be accurate we need to divide by the number of healthy validators.

**Review by sasha8 — APPROVED** [review]
We can think later how to make the chain health logic more accurate.

File: consensus/src/dag/dag_driver.rs [hunk 4]
```diff
@@ -290,6 +286,36 @@ impl DagDriver {
             prev_handle.abort();
         }
     }
+
+    fn calculate_payload_dimensions(&self, round: Round) -> (u64, u64) {
```
Comment (zekun000) [inline consensus/src/dag/dag_driver.rs:290]: I'd call it `payload_limit` instead of `payload_dimensions`

**Review by zekun000 — APPROVED** [review]

File: consensus/src/dag/dag_driver.rs (hunk 1 above)
Comment (ibalajiarun) [reply consensus/src/dag/dag_driver.rs:312]: Just in case, the division returns a 0, I want to pull at least one transaction.

**Pull request merged**